<a href="https://colab.research.google.com/github/daniela-lucio/on29-python-s17-s18-projeto-livre/blob/main/projeto_final_betania_daniela.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
from scipy.stats import ttest_1samp
import jupyter as jup


In [15]:
file = open('Base_Censo_Pop_Rua_2020.xlsx')

In [18]:
df = pd.read_excel("Base_Censo_Pop_Rua_2020.xlsx")
df

ID Rua_Acolhimento  Local_da_coleta_de_dados  \
0        0             Rua                       Rua   
1        1             Rua                       Rua   
2        2             Rua                       Rua   
3        3             Rua                       Rua   
4        4             Rua                       Rua   
...    ...             ...                       ...   
7267  7267       Acolhidos  Comunidades terapêuticas   
7268  7268       Acolhidos  Comunidades terapêuticas   
7269  7269       Acolhidos  Comunidades terapêuticas   
7270  7270       Acolhidos  Comunidades terapêuticas   
7271  7271       Acolhidos  Comunidades terapêuticas   

           Unidade_de_Acolhimento_US      Metodo          Turno        Data  \
0                                NaN  Entrevista          Manhã  26/10/2020   
1                                NaN  Entrevista          Manhã  26/10/2020   
2                                NaN  Entrevista          Manhã  26/10/2020   
3                                NaN  Entrevista          Manhã  26/10/2020   
4                                NaN  Entrevista          Manhã  26/10/2020   
...                              ...         ...            ...         ...   
7267  Instituto Revivendo com Cristo  Observação  Não se aplica         NaN   
7268  Instituto Revivendo com Cristo  Observação  Não se aplica         NaN   
7269  Instituto Revivendo com Cristo  Observação  Não se aplica         NaN   
7270  Instituto Revivendo com Cristo  Observação  Não se aplica         NaN   
7271  Instituto Revivendo com Cristo  Observação  Não se aplica         NaN   

            Bairro    AP  Codigo_da_RP  ... Frequencia_Maconha/Haxixe  \
0        Paciência  AP 5           5.3  ...             Não se aplica   
1     Campo Grande  AP 5           5.2  ...             Não se aplica   
2     Campo Grande  AP 5           5.2  ...             Não se aplica   
3     Campo Grande  AP 5           5.2  ...             Não se aplica   
4     Campo Grande  AP 5           5.2  ...             Não se aplica   
...            ...   ...           ...  ...                       ...   
7267     Guaratiba  AP 5           5.4  ...             Não se aplica   
7268     Guaratiba  AP 5           5.4  ...             Não se aplica   
7269     Guaratiba  AP 5           5.4  ...             Não se aplica   
7270     Guaratiba  AP 5           5.4  ...             Não se aplica   
7271     Guaratiba  AP 5           5.4  ...             Não se aplica   

      Frequencia_Crack/Similares Frequencia_Cocaina  \
0                  Não se aplica      Não se aplica   
1                  Não se aplica      Não se aplica   
2                  Não se aplica      Não se aplica   
3                  Não se aplica      Não se aplica   
4                  Não se aplica      Não se aplica   
...                          ...                ...   
7267               Não se aplica      Não se aplica   
7268               Não se aplica      Não se aplica   
7269               Não se aplica      Não se aplica   
7270               Não se aplica      Não se aplica   
7271               Não se aplica      Não se aplica   

      Frequencia_Inal_Cola_Solv_Tiner   Motivo_Droga Motivo_local_uso_droga  \
0                       Não se aplica  Não se aplica          Não se aplica   
1                       Não se aplica  Não se aplica          Não se aplica   
2                       Não se aplica  Não se aplica          Não se aplica   
3                       Não se aplica  Não se aplica          Não se aplica   
4                       Não se aplica  Não se aplica          Não se aplica   
...                               ...            ...                    ...   
7267                    Não se aplica  Não se aplica          Não se aplica   
7268                    Não se aplica  Não se aplica          Não se aplica   
7269                    Não se aplica  Não se aplica          Não se aplica   
7270                    Não se aplica  Não se aplica      

In [33]:
df_Base_Censo_Pop_Rua_2020_novo = df.drop(["Local_da_coleta_de_dados","Turno","Data","AP","Codigo_da_RP","RP","Codigo_da_RA","RA","Latitude","Longitude"], axis = 'columns')

In [34]:
df_Base_Censo_Pop_Rua_2020_novo.head()

ID Rua_Acolhimento Unidade_de_Acolhimento_US      Metodo        Bairro  \
0   0             Rua                       NaN  Entrevista     Paciência   
1   1             Rua                       NaN  Entrevista  Campo Grande   
2   2             Rua                       NaN  Entrevista  Campo Grande   
3   3             Rua                       NaN  Entrevista  Campo Grande   
4   4             Rua                       NaN  Entrevista  Campo Grande   

  Situacao_entrevista Motivo_situacao_impossivel Dormiu_na_rua_ultimos_7_dias  \
0            Possível              Não se aplica                          Sim   
1            Possível              Não se aplica                          Sim   
2            Possível              Não se aplica                          Sim   
3            Possível              Não se aplica                          Sim   
4            Possível              Não se aplica                          Sim   

  Respondeu_ao_questionario Questionario_de_Observacao  ...  \
0                       Sim                        Não  ...   
1                       Sim                        Não  ...   
2                       Sim                        Não  ...   
3                       Sim                        Não  ...   
4                       Sim                        Não  ...   

   Frequencia_Maconha/Haxixe Frequencia_Crack/Similares Frequencia_Cocaina  \
0              Não se aplica              Não se aplica      Não se aplica   
1              Não se aplica              Não se aplica      Não se aplica   
2              Não se aplica              Não se aplica      Não se aplica   
3              Não se aplica              Não se aplica      Não se aplica   
4              Não se aplica              Não se aplica      Não se aplica   

  Frequencia_Inal_Cola_Solv_Tiner   Motivo_Droga Motivo_local_uso_droga  \
0                   Não se aplica  Não se aplica          Não se aplica   
1                   Não se aplica  Não se aplica          Não se aplica   
2                   Não se aplica  Não se aplica          Não se aplica   
3                   Não se aplica  Não se aplica          Não se aplica   
4                   Não se aplica  Não se aplica          Não se aplica   

  Sabe_ler_escrever Frequentou_escola            Escolaridade  \
0               Sim   Sim, já estudei  Fundamental incompleto   
1               Sim   Sim, já estudei  Fundamental incompleto   
2               Sim   Sim, já estudei  Fundamental incompleto   
3               Sim   Sim, já estudei  Fundamental incompleto   
4               Sim   Sim, já estudei          Médio completo   

  Necessidade_sair_situacao  
0     Não quero sair da rua  
1                   Família  
2                   Moradia  
3     Não quero sair da rua  
4                     Outro  

[5 rows x 121 columns]

In [36]:
df_Base_Censo_Pop_Rua_2020_novo.dtypes

ID                            int64
Rua_Acolhimento              object
Unidade_de_Acolhimento_US    object
Metodo                       object
Bairro                       object
                              ...  
Motivo_local_uso_droga       object
Sabe_ler_escrever            object
Frequentou_escola            object
Escolaridade                 object
Necessidade_sair_situacao    object
Length: 121, dtype: object

In [37]:
df_Base_Censo_Pop_Rua_2020_novo[['Idade']].describe()

Idade
count  7272.000000
mean     72.222085
std     174.753777
min       0.000000
25%      30.000000
50%      40.000000
75%      52.000000
max     999.000000


Será preciso susbstiuir os valores 999 da columa idade por NaN (deixar a célula vazia), pois 999 representa "sem informação de idade".

In [38]:
file = open('Base_Censo_Pop_Rua_2022.xlsx')

In [39]:
df = pd.read_excel("Base_Censo_Pop_Rua_2022.xlsx")
df

ID  chave_unica  ano_censo Local_da_coleta_de_dados Rua_Instituicao  \
0        1         1315       2022                      Rua             Rua   
1        2         1316       2022                      Rua             Rua   
2        3         1317       2022                      Rua             Rua   
3        4         1318       2022                      Rua             Rua   
4        5         1319       2022                      Rua             Rua   
...    ...          ...        ...                      ...             ...   
7860  6629         6550       2022         Hospitais e CAPS   Institucional   
7861  6630         6551       2022         Hospitais e CAPS   Institucional   
7862  6631         6552       2022         Hospitais e CAPS   Institucional   
7863  6632         6553       2022         Hospitais e CAPS   Institucional   
7864  6633         6554       2022         Hospitais e CAPS   Institucional   

                        Nome_Instituicao Unidade_Provisoria_COVID19  \
0                          Não se aplica              Não se aplica   
1                          Não se aplica              Não se aplica   
2                          Não se aplica              Não se aplica   
3                          Não se aplica              Não se aplica   
4                          Não se aplica              Não se aplica   
...                                  ...                        ...   
7860          HOSPITAL ALBERT SCHWEITZER              Não se aplica   
7861               HOSPITAL SOUZA AGUIAR              Não se aplica   
7862                          UPA TIJUCA              Não se aplica   
7863  HOSPITAL DA MULHER MARISKA RIBEIRO              Não se aplica   
7864  HOSPITAL DA MULHER MARISKA RIBEIRO              Não se aplica   

                 Metodo    Bairro  Codigo_Bairro  ... consegue_agua_us  \
0     Entrevista direta    Centro              5  ...              Não   
1     Entrevista direta    Centro              5  ...              Não   
2     Entrevista direta    Centro              5  ...              Não   
3     Entrevista direta    Centro              5  ...              Não   
4     Entrevista direta    Centro              5  ...              Sim   
...                 ...       ...            ...  ...              ...   
7860         Observação  Realengo            139  ...              NaN   
7861         Observação    Centro              5  ...              NaN   
7862         Observação    Tijuca             33  ...              NaN   
7863         Observação     Bangu            141  ...              NaN   
7864         Observação     Bangu            141  ...              NaN   

      consegue_agua_igreja consegue_agua_projetos  consegue_agua_casa_amigos  \
0                      Não                    Não                        Não   
1                      Não                    Não                        Não   
2                      Não                    Não                        Não   
3                      Não                    Não                        Não   
4                      Não                    Não                        Não   
...                    ...                    ...                        ...   
7860                   NaN                    NaN                        NaN   
7861                   NaN                    NaN                        NaN   
7862                   NaN                    NaN                        NaN   
7863                   NaN                    NaN                        NaN   
7864                   NaN                    NaN                        NaN   

      consegue_agua_estacoes_transp consegue_agua_casa_desconhecidos  \
0                               Não                              Não   
1                               Não                              Não   
2                               Não                              Não   
3                               Não                              Não

In [40]:
df_Base_Censo_Pop_Rua_2022_novo = df.drop(["chave_unica","ano_censo","Local_da_coleta_de_dados", "Rua_Instituicao", "Nome_Instituicao","Unidade_Provisoria_COVID19", "Codigo_Bairro","RP","RA","Codigo_da_RA", "CAS","CAP","Subprefeitura","AP", "Latitude","Longitude"], axis = 'columns')

In [41]:
df_Base_Censo_Pop_Rua_2022_novo.head()

ID             Metodo  Bairro Motivo_situacao_impossivel  \
0   1  Entrevista direta  Centro                        NaN   
1   2  Entrevista direta  Centro                        NaN   
2   3  Entrevista direta  Centro                        NaN   
3   4  Entrevista direta  Centro                        NaN   
4   5  Entrevista direta  Centro                        NaN   

  Dormiu_na_rua_ultimos_7_dias  Idade Classificacao_idade       Sexo  \
0                          Sim   48.0              Adulto  Masculino   
1                          Sim   45.0              Adulto  Masculino   
2                          Sim   22.0              Adulto  Masculino   
3                          Sim   30.0              Adulto  Masculino   
4                          Sim   44.0              Adulto  Masculino   

      Genero Cor_raca  ... consegue_agua_us consegue_agua_igreja  \
0  Homem Cis    Parda  ...              Não                  Não   
1  Homem Cis    Preta  ...              Não                  Não   
2  Homem Cis    Preta  ...              Não                  Não   
3  Homem Cis    Parda  ...              Não                  Não   
4  Homem Cis    Parda  ...              Sim                  Não   

  consegue_agua_projetos consegue_agua_casa_amigos  \
0                    Não                       Não   
1                    Não                       Não   
2                    Não                       Não   
3                    Não                       Não   
4                    Não                       Não   

  consegue_agua_estacoes_transp consegue_agua_casa_desconhecidos  \
0                           Não                              Não   
1                           Não                              Não   
2                           Não                              Não   
3                           Não                              Não   
4                           Não                              Não   

  consegue_agua_construcao_civil consegue_agua_outros  \
0                            Não                  Não   
1                            Não                  Não   
2                            Não                  Não   
3                            Não                  Não   
4                            Não                  Não   

  consegue_agua_nao_consegue consegue_agua_ns_nr  
0                        Não                 Não  
1                        Não                 Não  
2                        Não                 Não  
3                        Não                 Não  
4                        Não                 Não  

[5 rows x 148 columns]

In [42]:
df_Base_Censo_Pop_Rua_2022_novo.dtypes

ID                                   int64
Metodo                              object
Bairro                              object
Motivo_situacao_impossivel          object
Dormiu_na_rua_ultimos_7_dias        object
                                     ...  
consegue_agua_casa_desconhecidos    object
consegue_agua_construcao_civil      object
consegue_agua_outros                object
consegue_agua_nao_consegue          object
consegue_agua_ns_nr                 object
Length: 148, dtype: object

In [43]:
df_Base_Censo_Pop_Rua_2022_novo[['Idade']].describe()

Idade
count  7865.000000
mean     46.131278
std      78.957749
min       0.000000
25%      30.500000
50%      40.500000
75%      50.000000
max     999.000000

Será preciso susbstiuir os valores 999 da columa idade por NaN (deixar a célula vazia), pois 999 representa "sem informação de idade".